pwndbg/commands/spray.py

```diff
@@ -0,0 +1,26 @@
+import argparse
+
+import gdb
+from pwnlib.util.cyclic import cyclic_gen
+
+import pwndbg.commands
+
+parser = argparse.ArgumentParser(description="Spray memory with a cyclic_gen() generated values")
+parser.add_argument("addr", help="Address to spray")
+parser.add_argument("length", nargs="?", help="Length of byte sequence", type=int, default=0)
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
+def spray(addr, length) -> None:
+    if length == 0:
+        try:
+            last_addr = next(p for p in pwndbg.gdblib.vmmap.get() if addr in p).end
+        except StopIteration:
+            last_addr = addr + 1  # Throw exception later when trying to write
```

Comment:
I think this would be more clear if it was just a for loop with a break, it's uncommon to use iterators like this.

Reply:
Fwiw:
```
$ rg 'next\(.*vmmap'
tests/gdb-tests/tests/heap/test_vis_heap_chunks.py
16:    heap_page = next(page for page in pwndbg.gdblib.vmmap.get() if page.objfile == "[heap]")

vis.patch
7:     heap_page = next(page for page in pwndbg.vmmap.get() if page.objfile == '[heap]')
```

Reply:
I think it's really the use of `StopIteration` that's the messy part here.